src/PowerShellEditorServices/Services/DebugAdapter/BreakpointService.cs

```diff
@@ -0,0 +1,412 @@
+                {
+                    if (int.TryParse(breakpoint.HitCondition, out int parsedHitCount))
+                    {
+                        hitCount = parsedHitCount;
+                    }
+                    else
+                    {
+                        breakpoint.Verified = false;
+                        breakpoint.Message = $"The specified HitCount '{breakpoint.HitCondition}' is not valid. " +
+                                              "The HitCount must be an integer number.";
+                        return null;
+                    }
+                }
+
+                // Create an Action scriptblock based on condition and/or hit count passed in.
+                if (hitCount.HasValue && string.IsNullOrWhiteSpace(breakpoint.Condition))
+                {
+                    // In the HitCount only case, this is simple as we can just use the HitCount
+                    // property on the breakpoint object which is represented by $_.
+                    string action = $"if ($_.HitCount -eq {hitCount}) {{ break }}";
+                    actionScriptBlock = ScriptBlock.Create(action);
+                }
+                else if (!string.IsNullOrWhiteSpace(breakpoint.Condition))
+                {
+                    // Must be either condition only OR condition and hit count.
+                    actionScriptBlock = ScriptBlock.Create(breakpoint.Condition);
+
+                    // Check for simple, common errors that ScriptBlock parsing will not catch
+                    // e.g. $i == 3 and $i > 3
+                    if (!ValidateBreakpointConditionAst(actionScriptBlock.Ast, out string message))
+                    {
+                        breakpoint.Verified = false;
+                        breakpoint.Message = message;
+                        return null;
+                    }
+
+                    // Check for "advanced" condition syntax i.e. if the user has specified
+                    // a "break" or  "continue" statement anywhere in their scriptblock,
+                    // pass their scriptblock through to the Action parameter as-is.
+                    Ast breakOrContinueStatementAst =
+                        actionScriptBlock.Ast.Find(
+                            ast => (ast is BreakStatementAst || ast is ContinueStatementAst), true);
+
+                    // If this isn't advanced syntax then the conditions string should be a simple
+                    // expression that needs to be wrapped in a "if" test that conditionally executes
+                    // a break statement.
+                    if (breakOrContinueStatementAst == null)
+                    {
+                        string wrappedCondition;
+
+                        if (hitCount.HasValue)
+                        {
+                            string globalHitCountVarName =
+                                $"$global:{s_psesGlobalVariableNamePrefix}BreakHitCounter_{breakpointHitCounter++}";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make the enclosing instance method 'static' or remove this set on the 'static' field.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4598462)